### ISSUE TITLE
Guest app shows empty file list

### ISSUE THREAD

**Issue by jesmrec** [owncloud/android#2167]
### Steps to reproduce
Create a guest account in the server by sharing with an email account,  and login with it in the Android client

### Current behaviour
File List is empty. iOS client shows the list in the same server

### Expected behaviour
File list is shown

Android 7 Nexus 6P v7.1.2

Server:

{"installed":true,"maintenance":false,"needsDbUpgrade":false,"version":"10.0.7.2","versionstring":"10.0.7","edition":"Community","productname":"ownCloud"}

**Comment by theScrabi** [comment]
The error I get when I try to access guest account is this

```
04-03 11:55:58.568 28120-28160/com.owncloud.android.debug E/ReadRemoteFolderOperation: Synchronized /: Unexpected exception
    java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.Object.toString()' on a null object reference
        at com.owncloud.android.lib.common.network.WebdavEntry.<init>(WebdavEntry.java:168)
        at com.owncloud.android.lib.resources.files.ReadRemoteFolderOperation.readData(ReadRemoteFolderOperation.java:142)
        at com.owncloud.android.lib.resources.files.ReadRemoteFolderOperation.run(ReadRemoteFolderOperation.java:89)
        at com.owncloud.android.lib.common.operations.RemoteOperation.runOperation(RemoteOperation.java:266)
        at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:142)
        at com.owncloud.android.operations.SynchronizeFolderOperation.fetchRemoteFolder(SynchronizeFolderOperation.java:253)
        at com.owncloud.android.operations.SynchronizeFolderOperation.run(SynchronizeFolderOperation.java:206)
        at com.owncloud.android.lib.common.operations.RemoteOperation.runOperation(RemoteOperation.java:266)
        at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:142)
        at com.owncloud.android.operations.common.SyncOperation.execute(SyncOperation.java:91)
        at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:143)
        at com.owncloud.android.lib.common.operations.RemoteOperation.runOperation(RemoteOperation.java:266)
        at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:232)
        at java.lang.Thread.run(Thread.java:764)
04-03 11:55:58.569 28120-28160/com.owncloud.android.debug E/SynchronizeFolderOperation: Checked chris.schabesberger@mailbox.org@docker.oc.solidgear.es:11780/ : Unexpected exception
    java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.Object.toString()' on a null object reference
        at com.owncloud.android.lib.common.network.WebdavEntry.<init>(WebdavEntry.java:168)
        at com.owncloud.android.lib.resources.files.ReadRemoteFolderOperation.readData(ReadRemoteFolderOperation.java:142)
        at com.owncloud.android.lib.resources.files.ReadRemoteFolderOperation.run(ReadRemoteFolderOperation.java:89)
        at com.owncloud.android.lib.common.operations.RemoteOperation.runOperation(RemoteOperation.java:266)
        at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:142)
        at com.owncloud.android.operations.SynchronizeFolderOperation.fetchRemoteFolder(SynchronizeFolderOperation.java:253)
        at com.owncloud.android.operations.SynchronizeFolderOperation.run(SynchronizeFolderOperation.java:206)
        at com.owncloud.android.lib.common.operations.RemoteOperation.runOperation(RemoteOperation.java:266)
        at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:142)
        at com.owncloud.android.operations.common.SyncOperation.execute(SyncOperation.java:91)
        at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:143)
        at com.owncloud.android.lib.common.operations.RemoteOperation.runOperation(RemoteOperation.java:266)
        at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:232)
        at java.lang.Thread.run(Thread.java:764)
```

**Comment by theScrabi** [comment]
fix provided here https://github.com/owncloud/android-library/pull/185

**Comment by jesmrec** [comment]
Fixed. Will be released in 2.8.0
